contracts/svm/testing/tests/swap.rs

```diff
@@ -0,0 +1,612 @@
+            svm,
+            &instructions,
+            &self.mint_authority,
+            &[&token_account, &self.mint_authority],
+        )
+        .unwrap();
+        token_account.pubkey()
+    }
+
+    pub fn create_mint(svm: &mut LiteSVM, token_program: Pubkey, decimals: u8) -> Self {
+        let mint = Keypair::new();
+        let mint_authority = generate_and_fund_key(svm);
+        let mint_rent = svm.minimum_balance_for_rent_exemption(spl_token_2022::state::Mint::LEN);
+        let instructions = vec![
+            system_instruction::create_account(
+                &mint_authority.pubkey(),
+                &mint.pubkey(),
+                mint_rent,
+                spl_token_2022::state::Mint::LEN.try_into().unwrap(),
+                &token_program,
+            ),
+            spl_token_2022::instruction::initialize_mint(
+                &token_program,
+                &mint.pubkey(),
+                &mint_authority.pubkey(),
+                None,
+                decimals,
+            )
+            .unwrap(),
+        ];
+        submit_transaction(
+            svm,
+            &instructions,
+            &mint_authority,
+            &[&mint, &mint_authority],
+        )
+        .unwrap();
+        Self {
+            mint: mint.pubkey(),
+            decimals,
+            mint_authority,
+            token_program,
+        }
+    }
+
+    pub fn get_amount_with_decimals(&self, amount: f64) -> u64 {
+        (amount * 10f64.powi(self.decimals as i32)).floor() as u64
+    }
+
+    pub fn get_associated_token_address(&self, owner: &Pubkey) -> Pubkey {
+        get_associated_token_address_with_program_id(owner, &self.mint, &self.token_program)
+    }
+}
+
+pub struct SwapSetupParams {
+    pub platform_fee_bps:      u64,
+    pub input_token_program:   Pubkey,
+    pub input_token_decimals:  u8,
+    pub output_token_program:  Pubkey,
+    pub output_token_decimals: u8,
+}
+
+pub struct SwapSetupResult {
+    pub svm:              LiteSVM,
+    pub trader:           Keypair,
+    pub searcher:         Keypair,
+    pub input_token:      Token,
+    pub output_token:     Token,
+    pub router_input_ta:  Pubkey,
+    pub router_output_ta: Pubkey,
+}
+
+pub fn setup_swap(args: SwapSetupParams) -> SwapSetupResult {
+    let SetupResult {
+        mut svm,
+        admin,
+        searcher,
+        ..
+    } = setup(None).expect("setup failed");
+
+    let trader = Keypair::new();
+    let input_token = Token::create_mint(
+        &mut svm,
+        args.input_token_program,
+        args.input_token_decimals,
+    );
+    let output_token = Token::create_mint(
+        &mut svm,
+        args.output_token_program,
+        args.output_token_decimals,
+    );
+
+    let set_swap_platform_fee_ix = set_swap_platform_fee_instruction(&admin, args.platform_fee_bps);
+    submit_transaction(&mut svm, &[set_swap_platform_fee_ix], &admin, &[&admin]).unwrap();
+
+    input_token.airdrop(&mut svm, &searcher.pubkey(), 10.);
```

Comment:
i kinda prefer the f64 notation cosmetically, but this is the king nit of nits

Reply:
will not address